lib/crawly/engine.ex

```diff
@@ -21,6 +21,23 @@ defmodule Crawly.Engine do
     GenServer.call(__MODULE__, {:start_spider, spider_name})
   end
 
+  @spec get_manager(module()) ::
+          pid() | {:error, :spider_non_exist} | {:error, :spider_not_found}
+  def get_manager(spider_name) do
+    case Map.fetch(running_spiders(), spider_name) do
+      :error ->
+        {:error, :spider_non_exist}
+
+      {:ok, pid_sup} ->
```

Comment:
I think you're getting a pid of the manager here, not a pid of supervisor. E.g. https://github.com/oltarasenko/crawly/blob/master/lib/crawly/engine.ex#L58

Reply:
not really, this is the ManagerSup

Reply:
true